endpoints/openrtb2/auction.go

```diff
@@ -1795,3 +1802,18 @@ func checkIfAppRequest(request []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+func (deps *endpointDeps) setIntegrationType(req *openrtb_ext.RequestWrapper) error {
+	reqExt, err := req.GetRequestExt()
+	if err != nil {
+		return err
+	}
+	reqPrebid := reqExt.GetPrebid()
+	if reqPrebid == nil {
+		reqPrebid = &openrtb_ext.ExtRequestPrebid{Integration: deps.cfg.AccountDefaults.DefaultIntegration}
+		reqExt.SetPrebid(reqPrebid)
+		req.RebuildRequest()
```

Comment:
Does it make sense to rebuild the request here? This function is called from `parseRequest` which is called from `Auction` and at the end of `Auction` we rebuild the request.

Reply:
I'll look into this, but I think you're right!